### DISCUSSION TITLE
escaping reserved characters

### DISCUSSION THREAD

**Discussion by randallm** [jsforce/jsforce#1501] [Ideas]
Hi, I noticed that users of the library currently have to manually escape `Connection#query(soql)` and `Connection#search` parameters. I built a simple sanitization npm module we use at @mixmaxhq in conjunction with with these calls: https://github.com/mixmaxhq/escapeSoqlAndSosl.

Is this something that would have a place inside the jsforce api?

**Comment by boutell** [comment]
Bump - any interest? How about supporting placeholders in `query` and passing an object with the values of the placeholders, using the ilbraqry @randallm created to auto-escape?

**Comment by techieshark** [comment]
Just to clarify, the escaping mentioned above is only currently missing when using `query(soql)` directly, right?

It looks like the safe alternative currently is "[By chaining other query construction methods, you can create a query programatically](https://jsforce.github.io/document/#using-query-method-chain)" which uses `escapeSOQLString` under the hood:

https://github.com/jsforce/jsforce/blob/62149865961f2950ee0fb595485cb83438e31fac/lib/soql-builder.js#L179
